src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -4003,23 +4027,24 @@ final boolean doExecute() throws SQLServerException {
     void onFedAuthInfo(SqlFedAuthInfo fedAuthInfo, TDSTokenHandler tdsTokenHandler) throws SQLServerException {
         assert (null != activeConnectionProperties.getProperty(SQLServerDriverStringProperty.USER.toString())
                 && null != activeConnectionProperties.getProperty(SQLServerDriverStringProperty.PASSWORD.toString()))
-                || ((authenticationString.trim().equalsIgnoreCase(
-                        SqlAuthentication.ActiveDirectoryIntegrated.toString()) && fedAuthRequiredPreLoginResponse));
+                || (authenticationString.trim().equalsIgnoreCase(SqlAuthentication.ActiveDirectoryIntegrated.toString())
+                        || authenticationString.trim().equalsIgnoreCase(SqlAuthentication.ActiveDirectoryMSI.toString())
+                                && fedAuthRequiredPreLoginResponse);
         assert null != fedAuthInfo;
 
         attemptRefreshTokenLocked = true;
-        fedAuthToken = getFedAuthToken(fedAuthInfo);
+        String accessToken = getFedAuthToken(fedAuthInfo);
         attemptRefreshTokenLocked = false;
 
         // fedAuthToken cannot be null.
-        assert null != fedAuthToken;
+        assert null != accessToken;
```

Comment:
change comment to reflect change to accessToken

Reply:
Code changed back to using FedAuthToken